Tests/Microsoft.Teams.Api.Tests/Clients/UserTokenClientTests.cs

```diff
@@ -0,0 +1,259 @@
+        {
+            Headers = { { "Custom-Header", "HeaderValue" } }
+        };
+
+        var mockHandler = new Mock<IHttpClient>();
+        mockHandler
+            .Setup(handler => handler.SendAsync<Token.Response>(It.IsAny<IHttpRequest>(), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new HttpResponse<Token.Response>
+            {
+                Headers = responseMessage.Headers,
+                StatusCode = HttpStatusCode.OK,
+                Body = new Token.Response
+                {
+                    ConnectionName = "connectionName",
+                    Token = "validToken"
+                }
+            });
+
+        var UserTokenClient = new UserTokenClient(mockHandler.Object);
+
+        var reqBody = await UserTokenClient.GetAsync(tokenRequest);
+
+        Assert.Equal("validToken", reqBody.Token);
+
+        string expecteUrl = "https://token.botframework.com/api/usertoken/GetToken?userId=userId-aad&connectionName=connectionName&channelId=webchat&code=200";
+        mockHandler.Verify(x => x.SendAsync<Token.Response>(It.Is<IHttpRequest>(arg => arg.Url == expecteUrl), It.IsAny<CancellationToken>()), Times.Once);
+    }
+
+    [Fact]
+    public async Task UserTokenClient_GetAadAsync_Async()
+    {
+        var aadTokenRequest = new GetAadTokenRequest
+        {
+            UserId = "userId-aad",
+            ConnectionName = "connectionName",
+            ChannelId = new ChannelId("webchat"),
+            ResourceUrls = ["resourceUrl1", "resourceUrl2"],
+        };
+
+        var responseMessage = new HttpResponseMessage
+        {
+            Headers = { { "Custom-Header", "HeaderValue" } }
+        };
+
+        IDictionary<string, Token.Response> addTokenResponses = new Dictionary<string, Token.Response>
+        {
+            {
+                "first",
+                new Token.Response
+                {
+                    ConnectionName = "connectionName",
+                    Token = "validToken"
+                }
+            },
+            {
+                "second",
+                new Token.Response
+                {
+                    ConnectionName = "connectionName",
+                    Token = "validToken"
+                }
+            }
+        };
+
+        var mockHandler = new Mock<IHttpClient>();
+        mockHandler
+            .Setup(handler => handler.SendAsync<IDictionary<string, Token.Response>>(It.IsAny<IHttpRequest>(), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(new HttpResponse<IDictionary<string, Token.Response>>
+            {
+                Headers = responseMessage.Headers,
+                StatusCode = HttpStatusCode.OK,
+                Body = addTokenResponses
+            });
+
+        var UserTokenClient = new UserTokenClient(mockHandler.Object);
+
+        var reqBody = await UserTokenClient.GetAadAsync(aadTokenRequest);
+
+        Assert.Equal(2, reqBody.Count);
+
+        // TODO - double check on ResourceURLs querystring
```

Comment:
Alex, pls confirm this resourceUrls query param is not passed correctly and will need to fix it

Reply:
updated